python/models/simple_tooltip_model.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sgtk
+
+shotgun_model = sgtk.platform.import_framework(
+    "tk-framework-shotgunutils",
+    "shotgun_model",
+)
+
+class SimpleTooltipModel(shotgun_model.SimpleShotgunModel):
+    """
+    A Shotgun model that sets simple tooltips for Shotgun entities.
+    """
+    def _set_tooltip(self, item, sg_item):
+        """
+        Sets the tooltip of the given item.
+
+        :param item:    Shotgun model item that requires a tooltip.
+        :param sg_item: Dictionary of the entity associated with the Shotgun
+                        model item.
+        """
+        item.setToolTip(
+            "%s: %s" % (
+                sg_item.get("type", "Entity Type Unknown"),
```

Comment:
If the name isn't set in Shotgun, the sg_item will probably look like this `{ ...., "code": None }`.
In this case the code will not work. 

The pattern i normally use is 

```
sg_item.get("code") or "Entity Name Unknown"
```

just to cover all cases.

The output feels very technical to me - examples:

- Shot: Foo Bar
- Entity Type Unknown: Foo Bar
- Shot: None
- Project: Entity Name Unknown
- CutomEntity13: Foo Bar
- Entity Type Unknown: Entity Name Unknown

I suggest:

- don't assume that `code` is the name field - use sg globals to determine the appropriate name field. Projects and tasks for example do not use `code`.
- the logic doesn't handle custom entities - again, use sg globals to resolve
- the output itself feels like it could be polished.

I suggest the following output (same cases as above)

- Shot: Foo Bar --> Shot Foo Bar
- Entity Type Unknown: Foo Bar --> This is an anomaly and no tooltip should be set - log a warning.
- Shot: None --> Unnamed Shot
- Project: Entity Name Unknown  --> Unnamed Project
- CutomEntity13: Foo Bar --> Level Foo Bar (Display name for CustomEntity13 is Level)
- Entity Type Unknown: Entity Name Unknown --> This is an anomaly and no tooltip should be set - log a warning.